Makefile.include

```diff
@@ -521,3 +521,19 @@ endif
 CFLAGS := $(patsubst -D%,,$(CFLAGS))
 CFLAGS := $(patsubst -U%,,$(CFLAGS))
 CFLAGS += -include '$(RIOTBUILD_CONFIG_HEADER_C)'
+
+# Set a custom channel if needed
+ifneq (,$(filter netdev2_ieee802154,$(USEMODULE)))
```

Comment:
I admit this is probably not the best check, maybe there's a more generic one ?
By the way, I still think that the internal block should go in a central place even it's example specific. I tested with the hello-world example on a samr21, this block is skipped. If the application requires network capabilities, it's executed.